ctapipe/image/ellipsoid.py

```diff
@@ -0,0 +1,443 @@
+    "CHEC": 0.5,
+    "DUMMY": 0,
+    "testcam": 0,
+}
+SPE_WIDTH = 0.5
+
+__all__ = [
+    "image_fit_parameters",
+    "ImageFitParameterizationError",
+    "PDFType",
+]
+
+
+class PDFType(Enum):
+    gaussian = "gaussian"
+    laplace = "laplace"
+    skewed = "skewed"
+
+
+def create_initial_guess(geometry, image, size):
+    """
+    This function computes the seeds of the fit with the Hillas parameters
+    Parameters
+    ----------
+    geometry: ctapipe.instrument.CameraGeometry
+        Camera geometry, the cleaning mask should be applied to improve performance
+    image: ndarray
+        Charge in each pixel, the cleaning mask should already be applied to
+        improve performance.
+    size: float/int
+        Total charge after cleaning and dilation
+    Returns
+    -------
+    initial_guess: dict
+    Seed parameters of the fit.
+    """
+    unit = geometry.pix_x.unit
+    hillas = hillas_parameters(geometry, image)  # compute Hillas parameters
+
+    initial_guess = {}
+
+    if unit.is_equivalent(u.m):
+        initial_guess["cog_x"] = hillas.x.to_value(unit)
+        initial_guess["cog_y"] = hillas.y.to_value(unit)
+    else:
+        initial_guess["x"] = hillas.fov_lon.to_value(unit)
+        initial_guess["y"] = hillas.fov_lat.to_value(unit)
+
+    initial_guess["length"] = hillas.length.to_value(unit)
+    initial_guess["width"] = hillas.width.to_value(unit)
+    initial_guess["psi"] = hillas.psi.value
+
+    initial_guess["skewness"] = hillas.skewness
+
+    if (hillas.width.to_value(unit) == 0) or (hillas.length.to_value(unit) == 0):
+        raise ImageFitParameterizationError("Hillas width and/or length is zero")
+
+    initial_guess["amplitude"] = size
+    return initial_guess
+
+
+def dilation(n, cleaned_mask, geometry):
+    """
+    This function adds n extra rows of pixels around the cleaned image
+    Parameters
+    ----------
+    n : int
+       number of extra rows to add after cleaning
+    cleaned_mask : ndarray
+       Cleaning mask (array of booleans) applied for Hillas parametrization
+    geometry : ctapipe.instrument.CameraGeometry
+        Camera geometry
+    Returns
+    -------
+    dilated_mask: ndarray of booleans
+    Cleaning mask after dilation
+    """
+    dilated_mask = cleaned_mask.copy()
+    for row in range(n):
+        dilated_mask = dilate(geometry, dilated_mask)
+
+    return dilated_mask
+
+
+def sensible_boundaries(geometry, cleaned_image, pdf):
+    """
+    Alternative boundaries of the fit based on the Hillas parameters.
+    Parameters
+    ----------
+    geometry: ctapipe.instrument.CameraGeometry
+        Camera geometry
+    cleaned_image: ndarray
+        Charge for each pixel, cleaning mask should be applied
+    pdf: PDFType instance
+        e.g. PDFType("gaussian")
+    Returns
+    -------
+    list of boundaries
+    """
+    hillas = hillas_parameters(geometry, cleaned_image)
+
+    unit = geometry.pix_x.unit
+    camera_radius = geometry.guess_radius()
```

Comment:
Is this memoized? If not it might be expensive to call this in the event loop

Reply:
it is current not memoized, so it will be quite slow to call this frequently. I didn't add a @lazyproperty to this call, since this is only one way to compute the edge of the camera (not sure it works well in all cases). It was just to give a quick way of computing a reasonable value.  Ideally we would have a `geom.radius` property that is  supplied externally like the pixel positions or telescope focal length, etc, and only guessed if none is provided.